### PR TITLE
Reformat work show page per #2050

### LINKED ISSUE #2050: Update and reorganize contents of Works show page
### Descriptive summary

Update the Works show page to align it with the overall current look-and-feel of Hyrax pages, to present the information is a slightly more compact form and to add a few additional metadata values to provide users with more basic information about the work.

Suggest this be implemented before #1033.

### Details

At the top of the page (below the breadcrumbs), a header is displayed with the word “Work” and a document icon, consistent with the header on the dashboard’s Works page.

Under the header, next to the work’s Title and status, the buttons for “Delete”, ‘Edit” and “Feature” are displayed in the upper right corner. The title is wrapped as necessary to accommodate these button. Since the Feature button is seen by a small subset of users (admins with the privileges to feature works), we might be able to use a wider column for the title by showing the Feature button on its own line, when needed.

In the section under the title, there is an area divided into two columns. On the left, a narrower column showing the work’s thumbnail (representative media) with a download link below. Under the download link, is a menu of citation options, and below that the social media badges. 

The wider column on the right includes the descriptive metadata and selected technical metadata. At the top of this column is the text entered in the work’s Abstract or Summary (dce:description). Under this text block is listed any other descriptive metadata values, with their corresponding display labels. The metadata also includes the date in which the work was last modified, as well as two pieces of information newly introduced to this screen: the total size of the work, and the total number of files.

Under this two-column section, the page is organized without columns. The Relationships section displays the name of the administrative set and any collections in which this work is a member. The labels are simplified to “Administrative set” and “Collection”.

Under the Relationships section, an “Items” table is shown with a row for each file (or nested work) in the work and includes the item name, date added, visibility level and action section menu.

work-show-revisions

### PR DESCRIPTION
**Pull request by aaron-collier**
Fixes #2050  ;

This adds a configurable work_type_icon (using font-awesome icons) and reformats the work show page as described in issue #2050.

Changes proposed in this pull request:
* Add configuration to enable display of icons on work show page (`display_work_icon`)
* Update template partials to reformat work show page
   * _attribute_rows - Add last modified
   * _citations - reformat links into pulldown
   * _metadata - remove border from table (sets table class to `borderless`)
   * _relationship - sets table class to `borderless`
   * _representative_media - sets thumbnail width to 250
   * _work_title - changes work title to h4 in panel header
   * _work_type - adds the human readable type and icon to the top of the page
   * show - reformats display into a bootstrap panel

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/views/hyrax/base/_attribute_rows.html.erb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<%= presenter.attribute_to_html(:date_modified, label: 'Last Modified') %>
```
Comment (jcoyne) [inline app/views/hyrax/base/_attribute_rows.html.erb]: What does this have to do with the rest of the PR?

File: app/assets/stylesheets/hyrax/_work-show.scss [hunk 2]
```diff
@@ -40,3 +40,30 @@ ul.tabular {
...
+  margin-top: -8px;
+}
+
+h4 {
+  line-height: 1.3;
+}
+
+.table > thead > tr > th, .table > thead > tr > td, .table > tbody > tr > th, .table > tbody > tr > td,
+.table > tfoot > tr > th, .table > tfoot > tr > td {
+  padding: 0px 8px 0px 8px;
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/assets/stylesheets/hyrax/_work-show.scss]: Is this overriding bootstrap? Is it possible to tweak a sass variable instead?

File: app/models/concerns/hyrax/work_type_icon.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      end
+    end
+
+    def work_type_icon
+      self.class.work_type_icon
+    end
+
+    def to_solr(solr_doc = {})
+      super(solr_doc).tap do |doc|
+        doc[Solrizer.solr_name('work_type_icon')] = work_type_icon
```
Comment (jcoyne) [inline app/models/concerns/hyrax/work_type_icon.rb]: Why does this need to be indexed? Should it it just be config? Should we have to reindeer everything if we wanted to change it?

File: app/views/hyrax/base/_attribute_rows.html.erb (hunk 1 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_attribute_rows.html.erb]: Based on the mockup, this is is moved to the attributes list from it's current location on the show template.

File: app/models/concerns/hyrax/work_type_icon.rb (hunk 3 above)
Comment (aaron-collier) [reply app/models/concerns/hyrax/work_type_icon.rb]: It may not need to be. I was following the human_readable_type example, but can test this without indexing. I'll follow up.

File: app/assets/stylesheets/hyrax/_work-show.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (aaron-collier) [reply app/assets/stylesheets/hyrax/_work-show.scss]: I'll take a look. I've not uses a sass variable, but happy to make the change.

File: app/models/concerns/hyrax/work_type_icon.rb (hunk 3 above)
Comment (jcoyne) [reply app/models/concerns/hyrax/work_type_icon.rb]: The human readable type is required because it's used (or was at it's addition)  as a facet.

File: app/views/hyrax/base/_attribute_rows.html.erb (hunk 1 above)
Comment (jcoyne) [reply app/views/hyrax/base/_attribute_rows.html.erb]: 👌

File: app/assets/stylesheets/hyrax/_work-show.scss [hunk 4]
```diff
@@ -40,3 +40,35 @@ ul.tabular {
   padding: $panel-body-padding;
   // text-align: center;
 }
+
+table.table.borderless > tbody > tr > td,
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/assets/stylesheets/hyrax/_work-show.scss]: Why do we need the leading selectors?  Could you do?
```
.table-borderless {
  th,
  td {
    border: 0;
  }
}
```

File: app/assets/stylesheets/hyrax/_work-show.scss [hunk 5]
```diff
@@ -40,3 +40,35 @@ ul.tabular {
...
+  line-height: 1.3;
+}
+
+.table > thead > tr > th,
+.table > thead > tr > td,
+.table > tbody > tr > th,
+.table > tbody > tr > td,
+.table > tfoot > tr > th,
+.table > tfoot > tr > td {
+  padding: 0px 8px;
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/assets/stylesheets/hyrax/_work-show.scss]: This looks like it would really squish a lot of the table spacing.

File: app/models/concerns/hyrax/work_type_icon.rb (hunk 3 above)
Comment (aaron-collier) [reply app/models/concerns/hyrax/work_type_icon.rb]: @jcoyne just realized I should be able to pull this from the i18n config. Doh! Will refactor with that in mind.

**Comment by hannahfrost** [comment]
@mjgiarlo No rush from my point of view. Thanks, all!

**Comment by aaron-collier** [comment]
@mjgiarlo @hannahfrost I'm perfectly happy waiting for this to merge. Just pushing the work we're doing upstream :-)

File: app/models/concerns/hyrax/work_type_icon.rb (hunk 3 above)
Comment (jcoyne) [reply app/models/concerns/hyrax/work_type_icon.rb]: Please don't use i18n for this, because you'd have to copy the value to every language (and it's unlikey to change based on language). Just the Hyrax::Configuration should be fine.

Comment (aaron-collier) [reply app/models/concerns/hyrax/work_type_icon.rb]: It's generated into the work_name.lang.yml file though. Seems like the best place to grab/set it..

File: app/assets/stylesheets/hyrax/_work-show.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (aaron-collier) [reply app/assets/stylesheets/hyrax/_work-show.scss]: You're right. I was seeing some odd padding earlier, but removing it was fine. Will be in a new push soon.

File: app/presenters/hyrax/work_show_presenter.rb [hunk 6]
```diff
@@ -138,6 +138,10 @@ def stats_path
       Hyrax::Engine.routes.url_helpers.stats_work_path(self)
     end
 
+    def model
+      solr_document['has_model_ssim'].first.underscore
```
Comment (jcoyne) [inline app/presenters/hyrax/work_show_presenter.rb]: Can we use `solr_document.to_model.model_name.singular` here instead?

File: app/views/hyrax/base/_metadata.html.erb [hunk 7]
```diff
@@ -1,8 +1,4 @@
-<h2><%= t('.header') %></h2>
-<table class="table table-striped <%= dom_class(presenter) %> attributes" <%= presenter.microdata_type_to_html %>>
-  <thead>
-    <tr><th><%= t('.attribute_name_label') %></th><th><%= t('.attribute_values_label') %></th></tr>
-  </thead>
+<table class="table-borderless <%= dom_class(presenter) %> attributes" <%= presenter.microdata_type_to_html %>>
```
Comment (jcoyne) [inline app/views/hyrax/base/_metadata.html.erb]: why are we dropping the `table` class?  why not `table table-borderless`?  If we had that, couldn't we drop a lot of custom styles?

File: app/views/hyrax/base/_metadata.html.erb [hunk 8]
```diff
@@ -1,8 +1,4 @@
-<h2><%= t('.header') %></h2>
```
Comment (jcoyne) [inline app/views/hyrax/base/_metadata.html.erb]: Should we remove this from the translations then?

File: app/views/hyrax/base/_metadata.html.erb [hunk 9]
```diff
@@ -1,8 +1,4 @@
...
-<table class="table table-striped <%= dom_class(presenter) %> attributes" <%= presenter.microdata_type_to_html %>>
-  <thead>
-    <tr><th><%= t('.attribute_name_label') %></th><th><%= t('.attribute_values_label') %></th></tr>
```
Comment (jcoyne) [inline app/views/hyrax/base/_metadata.html.erb]: remove these from the translation files?

File: app/views/hyrax/base/_relationships.html.erb [hunk 10]
```diff
@@ -1,5 +1,5 @@
 <h2><%= t('.header') %></h2>
-<table class="table table-striped relationships">
+<table class="table borderless relationships">
```
Comment (jcoyne) [inline app/views/hyrax/base/_relationships.html.erb]: is this the right class?

File: app/views/hyrax/base/_representative_media.html.erb [hunk 11]
```diff
@@ -1,6 +1,5 @@
 <% if presenter.representative_id.present? && presenter.representative_presenter.present? %>
   <%= media_display presenter.representative_presenter %>
 <% else %>
-  <%= image_tag 'default.png', class: "canonical-image" %>
+  <%= image_tag 'default.png', class: "canonical-image", width: 250 %>
```
Comment (jcoyne) [inline app/views/hyrax/base/_representative_media.html.erb]: Please add this to the stylesheets. It makes it easier to override downstream.

File: app/views/hyrax/base/_work_title.erb [hunk 12]
```diff
@@ -1,9 +1,16 @@
...
+      <% if index == 0 %>
+        <h4><%= title %>
+        <%= presenter.permission_badge %> <%= presenter.workflow.badge %>
+      </h4>
+      <% else %>
+          <h1><%= title %></h1>
+      <% end %>
+    </div>
+    <div class="col-sm-4 text-right">
+      <%= render "show_actions", presenter: @presenter %>
```
Comment (jcoyne) [inline app/views/hyrax/base/_work_title.erb]: Since we're using presenter as a local variable above can we use the local here too rather than the ivar?

File: app/views/hyrax/base/_work_type.html.erb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+<h3 class="work-type-tag">
+  <% if Hyrax.config.display_work_icon? %>
+    <span class="<%= t("hyrax.icons.#{presenter.model}") %>" aria-hidden="true"></span>
```
Comment (jcoyne) [inline app/views/hyrax/base/_work_type.html.erb]: Please don't put configuration in translations. This requires one change per language we support. Add it to configuration.

File: app/views/hyrax/base/_work_type.html.erb [hunk 14]
```diff
@@ -0,0 +1,6 @@
+<h3 class="work-type-tag">
+  <% if Hyrax.config.display_work_icon? %>
```
Comment (jcoyne) [inline app/views/hyrax/base/_work_type.html.erb]: Is there a use case for not displaying a work icon?

File: app/views/hyrax/file_sets/media_display/_image.html.erb [hunk 15]
```diff
@@ -4,14 +4,13 @@
       <%= image_tag thumbnail_url(file_set),
                     class: "representative-media",
                     alt: "",
-                    role: "presentation" %>
-      <%= link_to hyrax.download_path(file_set),
+                    role: "presentation",
+                    width: 250 %>
```
Comment (jcoyne) [inline app/views/hyrax/file_sets/media_display/_image.html.erb]: Please move this to stylesheets where it'll be easier for downstream consumer to override.

File: app/views/hyrax/base/_work_title.erb [hunk 16]
```diff
@@ -1,9 +1,16 @@
...
-        <h1>
-            <%= title %> <%= presenter.permission_badge %> <%= presenter.workflow.badge %>
-        </h1>
-    <% else %>
-        <h1><%= title %></h1>
-    <% end %>
+  <div class="row">
+    <div class="col-sm-8">
+      <% if index == 0 %>
+        <h4><%= title %>
```
Comment (jcoyne) [inline app/views/hyrax/base/_work_title.erb]: Why is this heading h4 here and h1 on line 9?  Why did we go to h4?  Is there already h1-h3 on the page?  The heading level is important for accessibility.

File: app/models/concerns/hyrax/solr_document/metadata.rb [hunk 17]
```diff
@@ -70,6 +70,7 @@ def self.coerce(input)
         attribute :mime_type, Solr::String, solr_name('mime_type', :stored_sortable)
         attribute :workflow_state, Solr::String, solr_name('workflow_state_name', :symbol)
         attribute :human_readable_type, Solr::String, solr_name('human_readable_type', :stored_searchable)
+        attribute :work_type_icon, Solr::String, solr_name('work_type_icon')
```
Comment (jcoyne) [inline app/models/concerns/hyrax/solr_document/metadata.rb]: I don't think there's an reason to continue declaring this in SolrDocument.

File: app/assets/stylesheets/hyrax/_work-show.scss [hunk 18]
```diff
@@ -40,3 +40,29 @@ ul.tabular {
...
+  color: #808080;
+}
+
+.work-type {
+  margin-bottom: 18px;
+  margin-top: -8px;
+}
+
+h4 {
+  line-height: 1.3;
```
Comment (jcoyne) [inline app/assets/stylesheets/hyrax/_work-show.scss]: This should be done by setting the bootstrap variable `$font-size-h4` rather than hard coding a value here.  In this way the downstream consumer has a way to override it.

**Comment by mjgiarlo** [comment]
@aaron-collier Now that 2.0.0, this is no longer blocked. As soon as you have time to respond to @jcoyne's feedback, we can get this approved.

File: app/views/hyrax/base/_metadata.html.erb (hunk 7 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_metadata.html.erb]: That was a mistake, I'll update that and fix the `borderless` to `table-borderless`, thanks for the catch.

File: app/views/hyrax/base/_work_type.html.erb (hunk 13 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_work_type.html.erb]: This is already in the translations (see: https://github.com/samvera/hyrax/blob/master/lib/generators/hyrax/work/templates/locale.en.yml.erb#L4)

Open to suggestion if I'm missing something here. :-)

Comment (jcoyne) [reply app/views/hyrax/base/_work_type.html.erb]: 😢

Comment (jcoyne) [reply app/views/hyrax/base/_work_type.html.erb]: Should this be `Sufia::ModelIcon.css_class_for(presenter.model)`

Comment (aaron-collier) [reply app/views/hyrax/base/_work_type.html.erb]: Assuming you mean `Hyrax::ModelIcon.css_class_for` right?

File: app/presenters/hyrax/work_show_presenter.rb (hunk 6 above)
Comment (aaron-collier) [reply app/presenters/hyrax/work_show_presenter.rb]: 👍

File: app/views/hyrax/base/_work_title.erb (hunk 12 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_work_title.erb]: Yeah, that was just a copy/paste from the show page when reformatting. Will fix.

File: app/views/hyrax/base/_work_type.html.erb (hunk 13 above)
Comment (jcoyne) [reply app/views/hyrax/base/_work_type.html.erb]: That's the one.

Comment (aaron-collier) [reply app/views/hyrax/base/_work_type.html.erb]: I think you are right here, but it brings up a couple of issues. The way things sit right now, I get an error that `model_name` doesn't doesn't exist for `GenericWork`. However, if I rename my `model` method to `model_name` I get a different error. I'm wondering if we should ticket that separately with more information, probably implementing a proper `model_name` method on works.

File: app/views/hyrax/base/_work_title.erb (hunk 16 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_work_title.erb]: I'm switching this up to h2 and the work_type to h1, I think this is probably what we need for accessibility.

File: app/views/hyrax/base/_metadata.html.erb (hunk 9 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_metadata.html.erb]: Done.

File: app/views/hyrax/base/_metadata.html.erb (hunk 8 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_metadata.html.erb]: Done.

File: app/views/hyrax/base/_attribute_rows.html.erb (hunk 1 above)
Comment (jcoyne) [reply app/views/hyrax/base/_attribute_rows.html.erb]: Can you show me where this is moved from? I'm not seeing it as a `-` in the diff.

File: app/views/hyrax/base/_citations.html.erb [hunk 19]
```diff
@@ -2,10 +2,14 @@
     <% if Hyrax.config.citations? %>
       <%= link_to "Citations", hyrax.citations_work_path(presenter), id: 'citations', class: 'btn btn-default' %>
     <% end %>
-    <div class='citations-heading'><%= t('.header') %></div>
```
Comment (jcoyne) [inline app/views/hyrax/base/_citations.html.erb]: Please remove `.citations-heading` and `.citations-list` from `app/assets/stylesheets/hyrax/_work-show.scss`

File: app/views/hyrax/base/_work_title.erb [hunk 20]
```diff
@@ -1,9 +1,16 @@
...
-        <h1>
-            <%= title %> <%= presenter.permission_badge %> <%= presenter.workflow.badge %>
-        </h1>
-    <% else %>
-        <h1><%= title %></h1>
-    <% end %>
+  <div class="row">
+    <div class="col-sm-8">
+      <% if index == 0 %>
+        <h2><%= title %>
```
Comment (jcoyne) [inline app/views/hyrax/base/_work_title.erb]: Do we have an `h1` on this page?  Not doing so would be an accessibility concern I believe.

File: app/views/hyrax/base/_attribute_rows.html.erb (hunk 1 above)
Comment (jcoyne) [reply app/views/hyrax/base/_attribute_rows.html.erb]: Ah, should this label be `t('hyrax.base.show.last_modified')` ?

File: app/views/hyrax/base/_work_type.html.erb [hunk 21]
```diff
@@ -0,0 +1,6 @@
+<h1 class="work-type-tag">
+  <% if Hyrax.config.display_work_icon? %>
+    <span class="<%= t("hyrax.icons.#{presenter.model}") %>" aria-hidden="true"></span>
```
Comment (jcoyne) [inline app/views/hyrax/base/_work_type.html.erb]: Isn't there a css_icon helper thing to use here?

File: app/views/hyrax/base/show.html.erb [hunk 22]
```diff
@@ -2,33 +2,40 @@
...
-    <%#= render 'user_activity', presenter: @presenter %>
+<div itemscope itemtype="http://schema.org/CreativeWork" class="row">
+  <div class="panel panel-default">
+    <div class="panel-heading">
+      <%= render 'work_title', presenter: @presenter %>
+    </div>
+    <div class="panel-body">
+      <div class="row">
+        <div class="col-sm-3 text-center">
+          <%# Use the fully qualified path to avoid RSpec error: translation missing: en.hyrax.base.show.html.erb.last_modified %>
```
Comment (jcoyne) [inline app/views/hyrax/base/show.html.erb]: I don't understand this comment. is it cruft?

File: app/views/hyrax/base/_attribute_rows.html.erb (hunk 1 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_attribute_rows.html.erb]: This wasn't moved. It was added to attribute_rows per the mock up, it was original just a value in the show page I believe. Can a translate title be used here? If so, I'll make that change later today.

File: app/views/hyrax/base/_citations.html.erb (hunk 19 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_citations.html.erb]: Will do

File: app/views/hyrax/base/_work_title.erb (hunk 20 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_work_title.erb]: We do, it's higher on the page in the _work_type partial.

File: app/views/hyrax/base/_work_type.html.erb (hunk 21 above)
Comment (aaron-collier) [reply app/views/hyrax/base/_work_type.html.erb]: I don't this so, not yet anyway. I was getting an error message with the helper syntax. I'll review adding one.

File: app/views/hyrax/base/show.html.erb (hunk 22 above)
Comment (aaron-collier) [reply app/views/hyrax/base/show.html.erb]: I think so. It's not mine. Maybe need to rebase, could have been removed after this.

Comment (jcoyne) [reply app/views/hyrax/base/show.html.erb]: I think the comment was referring to the line that followed it, which you removed.  Can you remove the comment too?

Comment (aaron-collier) [reply app/views/hyrax/base/show.html.erb]: Ah yes, this comment goes with the current source: https://github.com/samvera/hyrax/blob/master/app/views/hyrax/base/show.html.erb#L18 totally should be removed. Will do.

File: app/views/hyrax/base/_work_type.html.erb (hunk 21 above)
Comment (jcoyne) [reply app/views/hyrax/base/_work_type.html.erb]: `Hyrax::ModelIcon.css_class_for(presenter.model.class)` is what I'm looking for here.

File: app/views/hyrax/base/_work_type.html.erb [hunk 23]
```diff
@@ -0,0 +1,6 @@
+<h1 class="work-type-tag">
+  <% if Hyrax.config.display_work_icon? %>
```
Comment (jcoyne) [inline app/views/hyrax/base/_work_type.html.erb]: Do we need to support this conditional?  Can we just always show the work icon?

Comment (mjgiarlo) [reply app/views/hyrax/base/_work_type.html.erb]: Yeah, what's driving the need to make this configurable?

Comment (aaron-collier) [reply app/views/hyrax/base/_work_type.html.erb]: Overzealous cruft. Originally I was thinking the icon would be in the normal config, and added the option to enable it. Thinking through it (especially using the ModelIcon) it makes more sense to remove this and just override the view if someone wants to disable it. Will remove.
